src/main/java/uk/gov/hmcts/darts/arm/component/impl/ArmResponseFilesProcessSingleElementImpl.java

```diff
@@ -407,6 +408,14 @@ private void processUploadFileDataSuccess(ArmResponseUploadFileRecord armRespons
                 log.warn("Unable to verify annotation document checksum for external object {}", externalObjectDirectory.getId());
                 updateExternalObjectDirectoryStatus(externalObjectDirectory, armResponseChecksumVerificationFailedStatus);
             }
+        } else if (nonNull(externalObjectDirectory.getCaseDocument())) {
+            CaseDocumentEntity caseDocument = externalObjectDirectory.getCaseDocument();
+            if (nonNull(caseDocument.getChecksum())) {
```

Comment:
This checksum condition doesn't look necessary - if we're needing to use this column it would be easier to read if we reduce the redundant checks across these links?

```
                       Table "darts.case_document"
   Column    |           Type           | Collation | Nullable | Default 
-------------+--------------------------+-----------+----------+---------
 checksum    | character varying        |           | not null | 
```